packages/firestore/src/core/firestore_client.ts

```diff
@@ -291,32 +302,37 @@ export class FirestoreClient {
     });
 
     return Promise.resolve().then(() => {
-      this.persistence = new IndexedDbPersistence(
+      const persistence = new IndexedDbPersistence(
         storagePrefix,
         this.clientId,
         this.platform,
         this.asyncQueue,
         serializer
       );
-      if (WebStorageSharedClientState.isAvailable(this.platform)) {
-        this.sharedClientState = new WebStorageSharedClientState(
-          this.asyncQueue,
-          this.platform,
-          storagePrefix,
-          this.clientId,
-          user
-        );
-      } else {
+      this.persistence = persistence;
+
+      if (
+        settings.synchronizeTabs &&
+        !WebStorageSharedClientState.isAvailable(this.platform)
+      ) {
         if (process.env.USE_MOCK_PERSISTENCE !== 'YES') {
           throw new FirestoreError(
             Code.UNIMPLEMENTED,
             'IndexedDB persistence is only available on platforms that support LocalStorage.'
           );
         }
-        debug(LOG_TAG, 'Starting Persistence in test-only non multi-tab mode');
-        this.sharedClientState = new MemorySharedClientState();
       }
-      return this.persistence.start();
+
+      this.sharedClientState = settings.synchronizeTabs
+        ? new WebStorageSharedClientState(
```

Comment:
I'm confused about what's going on with USE_MOCK_PERSISTENCE.  It looks like this code used to fall back to MemorySharedClientState for mock persistence, but now it will always use WebStorageSharedClientState...  but the Code.UNIMPLEMENTED error condition above implies that WebStorageSharedClientState.isAvailable() will be false for node / USE_MOCK_PERSISTENCE.  So something seems out-of-sync.

I'm wondering if we don't need the memory fallback for mock persistence anymore since tests can explicitly opt out of synchronizeTabs, in which case we probably don't need the USE_MOCK_PERSISTENCE check above anymore?